src/antsibull/collection_meta_lint.py

```diff
@@ -0,0 +1,148 @@
+# Author: Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Ansible Project, 2024
+
+"""
+Classes to lint collection-meta.yaml
+"""
+
+from __future__ import annotations
+
+import os
+
+import pydantic as p
+from antsibull_core import app_context
+from antsibull_core.dependency_files import parse_pieces_file
+from antsibull_core.yaml import load_yaml_file
+
+from .collection_meta import CollectionMetadata, CollectionsMetadata, RemovalInformation
+
+
+class _Validator:
+    def __init__(self, *, all_collections: list[str], major_release: int):
+        self.errors: list[str] = []
+        self.all_collections = all_collections
+        self.major_release = major_release
+
+    def _validate_removal(
+        self, collection: str, removal: RemovalInformation, prefix: str
+    ) -> None:
+        if removal.version != "TBD" and removal.version <= self.major_release:
+            self.errors.append(
+                f"{prefix} version: Removal version {removal.version} must"
+                f" be larger than current major version {self.major_release}"
+            )
+
+        if (
+            removal.announce_version is not None
+            and removal.announce_version.major != self.major_release
+        ):
+            self.errors.append(
+                f"{prefix} announce_version: Major version of {removal.announce_version}"
+                f" must be the current major version {self.major_release}"
+            )
+
+        if removal.redirect_replacement_version is not None:
+            if removal.redirect_replacement_version <= self.major_release:
+                self.errors.append(
+                    f"{prefix} redirect_replacement_version: Redirect removal version"
+                    f" {removal.redirect_replacement_version} must be larger than"
+                    f" current major version {self.major_release}"
+                )
+            if (
+                removal.version != "TBD"
+                and removal.redirect_replacement_version >= removal.version
+            ):
+                self.errors.append(
+                    f"{prefix} redirect_replacement_version: Redirect removal version"
+                    f" {removal.redirect_replacement_version} must be smaller than"
+                    f" the removal major version {removal.version}"
+                )
+
+        if removal.reason == "renamed" and removal.new_name == collection:
+            self.errors.append(
+                f"{prefix} -> new_name: Must not be the collection's name"
```

Comment:
Doesn't prefix already end with `->` here?

```suggestion
                f"{prefix} new_name: Must not be the collection's name"
```